src/viadot/orchestration/prefect/utils.py

```diff
@@ -28,6 +28,394 @@
+        return date
+
+    def process_x_years_ago(self, year: str) -> str:
+        """
+        Retrieve the year of a date X years from now.
+
+        Args:
+            year (str): The year.
+
+        Returns:
+            str: A string containing the year of the specified time ago.
+        """
+        current_date = pendulum.now()
+        result = current_date.subtract(years=year).format("Y")
+        return result
+
+    def get_date_x_ago_full_date(self, number: int, unit: str) -> pendulum.DateTime:
+        """
+        Retrieve the full date for X units ago from today.
+
+        Args:
+            number (int): The number of units (years, months, days).
+            unit (str): The unit of time ('years', 'months', 'days').
+
+        Returns:
+            pendulum.DateTime: A date for X units ago from today.
+        """
+        full_date = {
+            "years": pendulum.now(self.dynamic_date_timezone).subtract(years=number),
+            "months": pendulum.now(self.dynamic_date_timezone).subtract(months=number),
+            "days": pendulum.now(self.dynamic_date_timezone).subtract(days=number),
+        }.get(unit)
+
+        return full_date
+
+    def create_date_dict(self) -> Dict[str, str]:
+        """
+        Create and return a key phrase: dynamic date value dictionary.
+        Dictionary values "today", "yesterday" and "last year previous month" are formatted into the dynamic_date_format.
+        The other values and their formatting:
+            - "this month" - A string date formatted with a string format '%m'.
+            - "last month" - A string date formatted with a format "%mm"
+            - "last year" - A string date formatted with a string format '%Y'
+            - "now time" - A string date formatted with a string format '%H%M%S'.
+
+        Returns:
+            Dict[str, str]: A dictionary with key phrases as keys and dynamically created dates as values.
+        """
+        today = pendulum.today(self.dynamic_date_timezone)
+        yesterday = pendulum.yesterday(self.dynamic_date_timezone)
+        last_month = today.subtract(months=1).month
+        last_year = today.subtract(years=1)
+        now_time = pendulum.now(self.dynamic_date_timezone)
+        last_day_prev_month = today.subtract(months=1).end_of("month")
+        replacements = {
+            "today": today.strftime(self.dynamic_date_format),
+            "yesterday": yesterday.strftime(self.dynamic_date_format),
+            "this month": today.strftime("%m"),
+            "last month": f"{last_month:02d}",
+            "this year": today.strftime("%Y"),
+            "last year": last_year.strftime("%Y"),
+            "now time": now_time.strftime("%H%M%S"),
+            "last day previous month": last_day_prev_month.strftime(
+                self.dynamic_date_format
+            ),
+        }
+        return replacements
+
+    def handle_singular_dates(self, match: List[tuple], key: str) -> pendulum.DateTime:
```

Comment:
mark this function internal by adding `_`.
```suggestion
    def _handle_singular_dates(self, match: List[tuple], key: str) -> pendulum.DateTime:
```